### ISSUE TITLE
points.buffer x/y should be phase/value

### ISSUE THREAD

**Issue by ramagottfried** [CNMAT/Music-and-Computing#54]

**Comment by ramagottfried** [comment]
@kulpajj what do you think about phase/value instead of x/y?
does that make more sense?

**Comment by equilet** [comment]
I know I'm not being asked, but I personally like it.

**Comment by ramagottfried** [comment]
which one?

**Comment by equilet** [comment]
I vote for both /phase and /value instead of /x and /y

**Comment by kulpajj** [comment]
/time and /value?

meh. I dunno.

**Comment by ramagottfried** [comment]
it's not time! : )
the timing depends on how fast the phase moves

**Comment by kulpajj** [comment]
I know I know I know. =)

Its /x and /y

**Comment by kulpajj** [comment]
The confusing thing about /phase is that in @normal_x 0, it IS /x, not
/phase.

**Comment by ramagottfried** [comment]
true... hmm. @equilet any thoughts on the @normal_x 0 issue? it's true that it's not "phase" in the traditional sense then.

**Comment by kulpajj** [comment]
I don't think anyone is confused by /x and /y

Its just a matter of getting to know the tools and the @normal_x modes.

**Comment by equilet** [comment]
What is 'normal x' in 'normal x mode'?  I'm confused.  ;)

**Comment by ramagottfried** [comment]
hmm, I sense some sarcasm there : )

by default @normal_x  is on (1) meaning that the /x values are normalized 0-1, if @normal_x is off (0) then the values are *not* normalized, so you would look them up (or interpolate) between them by the original value.

just in case : )

**Comment by equilet** [comment]
I see.  No, there was no sarcasm intended.  I'll take some time to review this with @kulpajj and get back into the thread.  Sorry if there was confusion introduced, I'm just trying to establish if the two use cases are different enough to come up with new terminology for the object, namespace, or both.

**Comment by ramagottfried** [comment]
it would be cool to have a name that makes it clearer that there is one value that is used as input for the lookup (/x) and anther that is the value that is output (/y).

thinking of it as points on an x/y plot makes some sense, but we don't at the moment have a way to lookup in the /y axis.

also x/y makes it seem like the points could be really 2D (i.e. jumping around in either axis), but actually the /x value should always be monotonically increasing so again it's not really truly plot.

thinking more about @normal_x 0, I think that is still ok to call it /phase since phase is really more about a point in cycle, or a transition between points. Normalization just helps with dealing with phase.

happy to hear your guys thoughts

**Comment by kulpajj** [comment]
I'm still partial to /x and /y.  It's very easy to understand.

/phase and /value is ok maybe, but I think saying /phase is an index in a
list isn't really true.  It's /lookup-x-position, so it's /x.

**Comment by ramagottfried** [comment]
hmm, /position and /value is pretty good... @equilet any thoughts?
/position is a lot of letters to type though... hmm, let's keep thinking about it... I have to run.

I like /x and /y too for the simplicity, but I think it's misleading for the reasons above...

what about:

/t and /value ?

`t` is often used as a variable in equations for interpolation which is what we're doing.
https://en.wikipedia.org/wiki/Linear_interpolation#Programming_language_support

I guess x/y is  like `f(x) = y` which is also what we're doing:
https://en.wikipedia.org/wiki/Linear_interpolation#Linear_interpolation_between_two_known_points

**Comment by ramagottfried** [comment]
but there's the problem of needing to specify x as increasing values (or I forget now, maybe I sort them internally).. ok have to run!

**Comment by kulpajj** [comment]
So /y is value.  Easy.  /x is a somewhat arbitrary value, and what we care
more about is its [[ index ]] is a lookup for the y value we want.  Harder
to communicate as a /name!  =-)  It seems like with this in mind, y could
be /value and x could be /index.  But that's pretty misleading when you
take a step back and think it doesn't really capture the real goings-on.
So what about /key and /value?

**Comment by kulpajj** [comment]
an alternative is staying with /x and /y but explicitly teaching what /x is
doing as a lookup value that gets a list-position, i.e. index, that
corresponds to a y value we want.  And that could be a whole tab in an m158
module that would clarify a lot for students.
